packages/components/src/navigation/item.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Icon, chevronRight } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import { useNavigationContext } from './context';
+import { ItemBadgeUI, ItemTitleUI, ItemUI } from './styles/navigation-styles';
+
+export default function NavigationItem( {
+	badge,
+	children,
+	className,
+	href,
+	item,
+	navigateToMenu,
+	onClick = noop,
+	title,
+	...props
+} ) {
+	const { activeItem, setActiveItem, setActiveMenu } = useNavigationContext();
+
+	const classes = classnames( 'components-navigation__item', className, {
+		'is-active': item && activeItem === item,
+	} );
+
+	const onItemClick = () => {
+		if ( navigateToMenu ) {
+			setActiveMenu( navigateToMenu );
+		} else if ( ! href ) {
+			setActiveItem( item );
+		}
+		onClick();
+	};
+
+	return (
+		<ItemUI className={ classes }>
+			<Button href={ href } onClick={ onItemClick } { ...props }>
+				{ title && (
+					<ItemTitleUI
+						className="components-navigation__item-title"
+						variant="body.small"
+						as="span"
+					>
+						{ title }
+					</ItemTitleUI>
+				) }
+
+				{ children }
+
+				{ badge && (
+					<ItemBadgeUI className="components-navigation__item-badge">
+						{ badge }
+					</ItemBadgeUI>
+				) }
+
+				{ navigateToMenu && <Icon icon={ chevronRight } /> }
```

Comment:
Noting here for a follow up PR because its the most visible place:

Checking presence of `navigateToMenu` for deciding if the item deserves a chevron or not presents awkwardness when programatically rendering `NavigationItem`s. Consider looping through a bunch of items, some of which are links with hrefs, some are custom components, and others are items with children.

```js
{ myItems.map( ( item ) => {
	return (
		<NavigationItem
			key={ item.id }
			item={ item.id }
			title={ item.title }
			href={ item.href }
			navigateToMenu={
				! item.href && ! item.customComponent && item.id
			}
		>
			{ item.customComponent ? item.customComponent : null }
		</NavigationItem>
	);
} ) }
```
The roundabout point I'm making is that it would be a nice-to-have if we do go the route of analyzing child elements that we could automate identifying which items have children and a chevron could be applied.

Reply:
Thanks for pointing this out!

We should definitely make some of these props safer, and automatize some behaviours.
If an item has `href` it shouldn't "activate" the navigation behaviour, but I wouldn't prevent items with children/custom components from being navigation items (see for example the WP logo item in the story).

Either way, I've noted this down, and it will be among the many follow up issues I'm going to open after merging this PR. 🙂